pallets/settlement/src/lib.rs

```diff
@@ -46,37 +60,33 @@ use polymesh_common_utilities::{
     SystematicIssuers::Settlement as SettlementDID,
 };
 use polymesh_primitives::{IdentityId, PortfolioId, Ticker};
-
-use codec::{Decode, Encode};
-use frame_support::weights::PostDispatchInfo;
-use frame_support::{
-    decl_error, decl_event, decl_module, decl_storage,
-    dispatch::{DispatchError, DispatchResult, DispatchResultWithPostInfo},
-    ensure, storage,
-    traits::Get,
-    weights::Weight,
-    IterableStorageDoubleMap, StorageHasher, Twox128,
-};
-use frame_system as system;
 use polymesh_primitives_derive::VecU8StrongTyped;
-use sp_runtime::traits::{Verify, Zero};
-use sp_runtime::DispatchErrorWithPostInfo;
+use sp_runtime::{
+    traits::{Dispatchable, Verify, Zero},
+    DispatchErrorWithPostInfo,
+};
 use sp_std::{collections::btree_set::BTreeSet, convert::TryFrom, prelude::*};
 
 type Identity<T> = identity::Module<T>;
+type System<T> = frame_system::Module<T>;
 
 pub trait Trait:
-    frame_system::Trait + CommonTrait + IdentityTrait + pallet_timestamp::Trait
+    frame_system::Trait + CommonTrait + IdentityTrait + pallet_timestamp::Trait + scheduler::Trait
 {
     /// The overarching event type.
     type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
     /// Asset module
     type Asset: AssetTrait<Self::Balance, Self::AccountId>;
-    /// The maximum number of total legs in scheduled instructions that can be executed in a single block.
-    /// Any excess instructions are scheduled in later blocks.
-    type MaxScheduledInstructionLegsPerBlock: Get<u32>;
     /// The maximum number of total legs allowed for a instruction can have.
     type MaxLegsInAInstruction: Get<u32>;
+    /// Scheduler of timelocked bridge transactions.
+    type Scheduler: ScheduleAnon<Self::BlockNumber, Self::SchedulerCall, Self::SchedulerOrigin>;
+    /// A type for identity-mapping the `Origin` type. Used by the scheduler.
+    type SchedulerOrigin: From<RawOrigin<Self::AccountId>>;
```

Comment:
can't we directly use something like this 

```
type Scheduler: ScheduleAnon<Self::BlockNumber, <Self as frame_system::Trait>::Call, <Self as frame_system::Trait>::Origin>;
```

Reply:
Unfortunately that's not possible due to the missing trait instances. The origin should be `OriginCaller` rather than the origin type used in dispatchables. Also there is no `From` `Self::Call` to `<T as frame_system::Trait>::Call`.